### PR TITLE
ConservedHeadersJettyErrorHandler

### PR DESCRIPTION
**Pull request by dkaukov**
Preserves conserved headers for unhandled servlet exceptions.

### CODE REVIEW

**Comment by mikebell90** [comment]
To be clear @dkaukov these were all concerns Scott and I came up with and feel need addressing.

File: otj-server-jaxrs/src/test/java/com/opentable/server/TestJaxRsServerConfiguration.java [hunk 1]
```diff
@@ -98,6 +99,13 @@ public Response get5xx() {
             throw new RuntimeException("it has happened");
         }
 
+        @GET()
+        @Path("5xx2")
+        public Response get5xx2() {
+            LOG.info("It is going to happen.");
+            throw new InternalServerErrorException("it has happened");
```
[the remark this answers is not in the document]
Comment (dkaukov) [reply otj-server-jaxrs/src/test/java/com/opentable/server/TestJaxRsServerConfiguration.java]: @scottjohnson The handler is "chained", so all user defined handlers will be called. There are existing test for it in jax-rs server. Also there are test for `RuntimeException`

File: otj-server-core/src/main/java/com/opentable/server/OTJettyServletWebServerFactory.java [hunk 2]
```diff
@@ -50,6 +50,7 @@ public OTJettyServletWebServerFactory() {
     protected org.eclipse.jetty.webapp.Configuration[] getWebAppContextConfigurations(WebAppContext webAppContext,
                                                                                       ServletContextInitializer... initializers) {
         webAppContextCustomizers.ifPresent(consumers -> consumers.forEach(c -> c.accept(webAppContext)));
+        webAppContext.setErrorHandler(new ConservedHeadersJettyErrorHandler(webAppContext.getErrorHandler()));
```
[the remark this answers is not in the document]
Comment (dkaukov) [reply otj-server-core/src/main/java/com/opentable/server/OTJettyServletWebServerFactory.java]: It is "chained", that is wht 1st parameter of constructor is "old" error handler.

File: otj-server-core/src/main/java/com/opentable/server/ConservedHeadersJettyErrorHandler.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ConservedHeadersJettyErrorHandler(ErrorHandler delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
+        if (this.delegate != null) {
+            ConservedHeadersFilter.extractHeaders(request)
+                .forEach((header, value) -> response.setHeader(header.getHeaderName(), value));
+            this.delegate.handle(target, baseRequest, request, response);
```
[the remark this answers is not in the document]
Comment (dkaukov) [reply otj-server-core/src/main/java/com/opentable/server/ConservedHeadersJettyErrorHandler.java]: Error handlers are called after headers are cleaned in DISPATCH_ERROR branch

**Comment by mikebell90** [comment]
Looks much better
Would still like to 
* Examine server/stack trace issue
* Justify rethrown exception in exception filter.

**Comment by dkaukov** [comment]
@mikebell90 @scottjohnson 

Regarding re-throwing exception in `ExceptionLogFilter`.. My thinking was following:
The `doFilter()` declared as:
```@Override
        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {```

which means that exception thrown there is already "wrapped" and different from original thrown exception. Re-wrapping it to the `NestedServletException` with original cause should not change execution flow and we not loosing information.

**Pull request merged**